californium-core/pom.xml

```diff
@@ -37,6 +37,35 @@
 	<build>
 		<plugins>
 			<plugin>
+				<groupId>org.apache.felix</groupId>
+				<artifactId>maven-bundle-plugin</artifactId>
+				<extensions>true</extensions>
+				<configuration>
+					<instructions>
+						<Export-Package>
+							org.eclipse.californium.core,
```

Comment:
Yes and no. 

It is correct that the Apache HttpComponents used in the build are not provided as OSGi bundles directly. But there is also an OSGi version available (I really don't understand why there are two different versions). Well at least httpcore and httpclient are provided as OSGi bundles. http-nio seems not to be available as OSGi version. That means users that want to use californium-proxy will have an issue in an OSGi runtime because of the missing http-nio bundle. They will need to create an OSGi-ified version themselves. IMHO not a good practice by Apache HttpComponents that need to be discussed with the HttpComponents team.

Nevertheless the used dependency from Maven does not contain OSGi meta-data as you already noticed. We could either change the dependency to the OSGi versions of that bundles (I really have no clue why they have two versions) or keep it that way.

Keeping it that way simply means we have no lower version bound. So it is only checked if that package is provided by anybody. IMHO better than nothing. We just can't tell at runtime which bundle is wired if multiple versions are available in the runtime, and we can't specify the version we need at minimum.